File: quick_sort.cpp
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
#define N 100000 // number of data to be sort

void swap(int *a,int *b){
	int h = *a;
	*a = *b;
	*b = h;
}

int partition(int a[],int p,int r){
	int i,j,x;
	i = p;
	x = a[r];
	for(j=p;j<r;j++){
		if(a[j]<=x){
			swap(&a[j],&a[i]);
			i++;
		}
	}
	swap(&a[i],&a[r]);
	return i;
}

void QuickSort(int a[],int p,int r){
	int q;
	if(p<r){
		q = partition(a,p,r); // partitioning Array a[]
		QuickSort(a,p,q-1);
		QuickSort(a,q+1,r);
	}
}

int main(){
	srand(time(NULL));
	int i,a[N];
	double start,stop;
	for(i=0;i<N;i++) a[i] = rand()%100;
	//for(i=0;i<N;i++) printf("%d ",a[i]); // printing data before sorting
	printf("\n");
	start = clock();
	quickSort(a,0,N-1);
	stop = clock();
	//for(i=0;i<N;i++) printf("%d ",a[i]); // printing data after sorting
	printf("\n\nRequired time to Sort : %.3lf",(stop-start)/CLOCKS_PER_SEC);
	return 0;
}

File: quick_sort.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import random,time

N = 10000 # amount of data to be sort
a = []

def swap(x,y):
	return y,x

def partition(a,p,r):
	x = a[r]
	i = p-1
	j = p
	while j<r:
		if a[j] <= x:
			i+=1
			a[i],a[j] = swap(a[i],a[j])
		j+=1
	a[i+1],a[r] = swap(a[i+1],a[r])
	return i+1

def quick_sort(a,p,r):
	if p < r:
		q = partition(a,p,r)
		quick_sort(a,p,q-1)
		quick_sort(a,q+1,r)

def gen_value():
	i=0
	while i<N:
		a.append(random.randint(0,100))
		i+=1

def Display():
	i=0
	while i<N:
		print a[i],' ',
		i+=1
	print '\n'

gen_value()
# Display()
start = time.clock()
quick_sort(a,0,N-1)
stop = time.clock()
# Display()

print 'Required time to sort : ',(stop - start)
